src/internal.rs

```diff
@@ -798,13 +839,26 @@ pub(crate) struct Rav1dFrameContext_frame_thread_progress {
 }
 
 pub(crate) struct Rav1dFrameContext {
+    /// Index in [`Rav1dContext::fc`]
+    pub index: usize,
+
     pub data: RwLock<Rav1dFrameData>,
     pub in_cdf: RwLock<CdfThreadContext>,
     pub task_thread: Rav1dFrameContext_task_thread,
     pub frame_thread_progress: Rav1dFrameContext_frame_thread_progress,
 }
 
 impl Rav1dFrameContext {
+    pub(crate) unsafe fn zeroed(index: usize) -> Self {
+        Self {
+            index,
+            data: RwLock::new(unsafe { Rav1dFrameData::zeroed() }),
+            in_cdf: RwLock::new(CdfThreadContext::default()),
+            task_thread: Rav1dFrameContext_task_thread::default(),
+            frame_thread_progress: Rav1dFrameContext_frame_thread_progress::default(),
```

Comment:
```suggestion
            in_cdf: Default::default(),
            task_thread: Default::default(),
            frame_thread_progress: Default::default(),
```

Can we just do this?  Is there a reason you did it with the specific types?